api/files.py

```diff
@@ -34,9 +34,20 @@ def getHashingFieldStorage(upload_dir, hash_alg):
     class HashingFieldStorage(cgi.FieldStorage):
         bufsize = 2**20
         def make_file(self, binary=None):
+            if not self.filename:
+                return self.file
```

Comment:
Only if the form field has a filename we treat that as a file.
e.g. we are not creating a file and a hash for the form metadata.